socorro/external/boto/connection_context.py

```diff
@@ -242,3 +246,29 @@ def head_object(self, bucket, key):
                     f"(bucket={bucket!r} key={key}) not found, no value returned"
                 ) from exc
             raise
+
+    def delete_object(self, bucket, key):
+        """DELETE action on an object in a S3 bucket
+
+        :arg bucket: the name of the bucket
+        :arg key: the key for the object to DELETE
+
+        :returns: S3 DELETE response
+
+        :raises KeyNotFound: if the object doesn't exist
+        :raises botocore.exceptions.ClientError: connection issues, permissions
+            issues, bucket is missing, etc.
+
+        """
+        try:
+            return self.client.delete_object(Bucket=bucket, Key=key)
+        except self.client.exceptions.NoSuchKey as exc:
+            raise KeyNotFound(
+                f"(bucket={bucket!r} key={key}) not found, no value returned"
+            ) from exc
+        except self.client.exceptions.ClientError as exc:
+            if "404" in str(exc):
```

Comment:
Is there a more specific substring we could look for? Maybe "http: 404" or something like that? If the msg ever included a crash id, those could have a 404 in them.

Having said that, this code isn't likely to be used and the S3 stuff is leaving soon anyhow, so we don't _need_ to look into this. But if we knew offhand, it'd be good to improve it.